nvme.c

```diff
@@ -5888,6 +5888,8 @@ static int set_feature(int argc, char **argv, struct command *cmd, struct plugin
 							cfg.file, strerror(errno));
 					return -EINVAL;
 				}
+			} else {
+				ffd = STDIN_FILENO;
```

Comment:
Seems the stdin stream will be closed by the close_file() for cleanup. Is it expected?

Reply:
Ah, no, sorry. Will fix.

Reply:
Why not just dup the `stdin` so we just can use the common close path.

Reply:
Ignore my comment. I though it's something different. This looks good.

There is still some more cleanup stuff to do with the auto close stuff, e.g. `submit_io` could need some cleanup and I thought we were discussing this here.

Reply:
Hmm, but I still prefer this here:

```diff
-                       if (strlen(cfg.file)) {
+                       if (strlen(cfg.file))
                                ffd = open(cfg.file, O_RDONLY);
-                               if (ffd <= 0) {
-                                       nvme_show_error("Failed to open file %s: %s",
-                                                       cfg.file, strerror(errno));
-                                       return -EINVAL;
-                               }
-                               err = read(ffd, buf, cfg.data_len);
-                       } else {
-                               err = read(STDIN_FILENO, buf, cfg.data_len);
+                       else
+                               ffd = dup(STDIN_FILENO);
+
+                       if (ffd < 0) {
+                               nvme_show_error("Failed to open file %s: %s",
+                                               cfg.file, strerror(errno));
+                               return -EINVAL;
                        }
+
+                       err = read(ffd, buf, cfg.data_len);
                        if (err < 0) {
                                nvme_show_error("failed to read data buffer from input file: %s",
                                                strerror(errno));
```

Could you update the patch this way? Thanks!

Reply:
Done!